sdk/tables/data-tables/src/TableServiceClient.ts

```diff
@@ -224,56 +211,55 @@ export class TableServiceClient {
    * @param properties - The Table Service properties.
    * @param options - The options parameters.
    */
-  public async setProperties(
+  public setProperties(
     properties: ServiceProperties,
     options: SetPropertiesOptions = {}
   ): Promise<SetPropertiesResponse> {
-    const { span, updatedOptions } = createSpan("TableServiceClient-setProperties", options);
-    try {
-      return await this.service.setProperties(properties, updatedOptions);
-    } catch (e) {
-      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
-      throw e;
-    } finally {
-      span.end();
-    }
+    return tracingClient.withSpan("TableServiceClient.setProperties", options, (updatedOptions) =>
+      this.service.setProperties(properties, updatedOptions)
+    );
   }
 
   /**
    * Creates a new table under the given account.
    * @param name - The name of the table.
    * @param options - The options parameters.
    */
-  public async createTable(name: string, options: OperationOptions = {}): Promise<void> {
-    const { span, updatedOptions } = createSpan("TableServiceClient-createTable", options);
-    try {
-      await this.table.create({ name }, { ...updatedOptions });
-    } catch (e) {
-      handleTableAlreadyExists(e, { ...updatedOptions, span, logger, tableName: name });
-    } finally {
-      span.end();
-    }
+  public createTable(name: string, options: OperationOptions = {}): Promise<void> {
+    return tracingClient.withSpan(
+      "TableServiceClient.createTable",
+      options,
+      async (updatedOptions) => {
+        try {
+          await this.table.create({ name }, updatedOptions);
+        } catch (e) {
+          handleTableAlreadyExists(e, { ...updatedOptions, logger, tableName: name });
+        }
+      }
+    );
   }
 
   /**
    * Operation permanently deletes the specified table.
    * @param name - The name of the table.
    * @param options - The options parameters.
    */
-  public async deleteTable(name: string, options: OperationOptions = {}): Promise<void> {
-    const { span, updatedOptions } = createSpan("TableServiceClient-deleteTable", options);
-    try {
-      await this.table.delete(name, updatedOptions);
-    } catch (e) {
-      if (e.statusCode === 404) {
-        logger.info("TableServiceClient-deleteTable: Table doesn't exist");
-      } else {
-        span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
-        throw e;
+  public deleteTable(name: string, options: OperationOptions = {}): Promise<void> {
+    return tracingClient.withSpan(
+      "TableServiceClient.deleteTable",
+      options,
+      async (updatedOptions) => {
+        try {
+          await this.table.delete(name, updatedOptions);
+        } catch (e) {
+          if (e.statusCode === 404) {
+            logger.info("TableServiceClient-deleteTable: Table doesn't exist");
```

Comment:
For the logger info too, should we use a "." instead of a "-" as the convention, if that is what we are standardizing on across our spans for all packages? @maorleger

Reply:
Yes, fixed. Thank you!